src/flask/app.py

```diff
@@ -1581,7 +1579,7 @@ def ensure_sync(self, func: t.Callable) -> t.Callable:
         .. versionadded:: 2.0
         """
         if iscoroutinefunction(func):
-            return run_async(func)
+            return async_to_sync(func)
```

Comment:
It still seems like the `async_to_sync` implementation is the only thing users will want to customize, rather than this entire `ensure_sync` method. The current design still requires users to duplicate this `iscoroutinefunction(...)` boilerplate. But it would be just as easy to parameterize the `async_to_sync` function instead, no?

Reply:
Done, added `async_to_sync` as an overrideable method.